BraintreeDropIn/BTCardFormViewController.m

```diff
@@ -682,30 +682,32 @@ - (void)enrollCard {
 #pragma mark FormField Delegate Methods
 
 - (void)validateButtonPressed:(__unused BTUIKFormField *)formField {
-    BTUIKCardType *cardType = self.cardNumberField.cardType;
-    BOOL cardSupported = NO;
-    for (id object in self.supportedCardTypes) {
-        BTUIKPaymentOptionType supportedCardType = ((NSNumber*)object).intValue;
-        if ( supportedCardType == [BTUIKViewUtil paymentMethodTypeForCardType:cardType]) {
-            cardSupported = YES;
-            break;
-        }
-    }
+    NSNumber *cardType = [NSNumber numberWithInt:[BTUIKViewUtil paymentMethodTypeForCardType:self.cardNumberField.cardType]];
+    BOOL cardSupported = [self.supportedCardTypes containsObject:cardType];
+
     if (!cardSupported) {
         [self cardNumberErrorHidden:NO errorString:BTUIKLocalizedString(CARD_NOT_ACCEPTED_ERROR_LABEL)];
         return;
     }
-    
-    if (!self.unionPayEnabledMerchant) {
-        [self cardNumberErrorHidden:formField.valid];
-        if (formField.valid) {
-            self.cardNumberField.state = BTUIKCardNumberFormFieldStateDefault;
-            self.collapsed = NO;
-            [self advanceFocusFromField:formField];
-        }
-    } else {
+
+    if (!formField.valid) {
+        [self cardNumberErrorHidden:NO];
+        return;
+    }
+
+    if (!self.configuration) {
+        self.cardNumberField.state = BTUIKCardNumberFormFieldStateLoading;
+        return;
+    }
```

Comment:
This is the main addition to this method -- if we don't yet have the configuration, show the loading indicator next to the card number form instead of trying to advance to the next field.